File: zdg_tasks_frontend/lib/shared/models/receipt.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'attachment.dart';

part 'receipt.freezed.dart';
part 'receipt.g.dart';

@freezed
abstract class Receipt with _$Receipt {
  const factory Receipt({
    required int id,
    // Integer minor units (ngwee).
    required int amount,
    required bool verified,
    DateTime? verifiedAt,
    Attachment? attachment,
  }) = _Receipt;

  factory Receipt.fromJson(Map<String, dynamic> json) => _$ReceiptFromJson(json);
}

File: zdg_tasks_backend/app/Models/Receipt.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

#[Fillable([
    'task_id', 'attachment_id', 'amount', 'verified', 'verified_by',
    'verified_at',
])]
class Receipt extends Model
{
    /**
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            // Integer minor units (ngwee).
            'amount' => 'integer',
            'verified' => 'boolean',
            'verified_at' => 'datetime',
        ];
    }

    public function task(): BelongsTo
    {
        return $this->belongsTo(Task::class);
    }

    public function attachment(): BelongsTo
    {
        return $this->belongsTo(Attachment::class);
    }

    public function verifier(): BelongsTo
    {
        return $this->belongsTo(User::class, 'verified_by');
    }
}
